### PR TITLE
Feature/tracing

### PR DESCRIPTION
**Pull request by csucu**
- added opentelemetery tracing

### CODE REVIEW

File: tracer/trace.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/propagators"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/semconv"
+)
+
+// InitTracingProvider creates a new otel tracing provide and returns a closer
```
Comment (amnonbc) [inline tracer/trace.go:19]: godock comments should end with a fullstop.

**Review by amnonbc — APPROVED** [review]

File: tracer/trace.go (hunk 1 above)
Comment (csucu) [reply tracer/trace.go]: done

**Comment by csucu** [comment]
> This test will sleep for a total of 2 minutes 4 seconds!
> 
> We need tests to run fast - in a couple of milliseconds at most.
> So we should reduce the sleep times, or use a mockable time library.

That test file was supposed tobe untracked, i've removed it for now though. I'll add some proper test files soon

File: tracer/trace.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	emptyTraceID = trace.ID{}
+	emptySpanID = trace.SpanID{}
+)
+
+type Tracer struct {
+	trace.Tracer
+}
+
+// InitTracingProvider creates a new otel tracing provider and returns it with a closer.
+func InitTracingProvider(collectorAddress string, serviceName string) (*Tracer, func(), error) {
```
Comment (amnonbc) [inline tracer/trace.go:31]: three return values is quite a lot.

Can't we just add a close method to Tracer?

Comment (csucu) [reply tracer/trace.go:31]: yea, thats should be fine, will add

Comment (csucu) [reply tracer/trace.go]: done

**Review by amnonbc — COMMENTED** [review]
don't you need to rename the directory too?

**Comment by dmotylev** [comment]
obsolete

**Pull request closed without merging**